crates/rnote-engine/src/strokes/vectorimage.rs

```diff
@@ -156,21 +160,37 @@ impl VectorImage {
         let mut svg_tree = usvg::Tree::from_str(svg_data, &usvg::Options::default())?;
         svg_tree.convert_text(&render::USVG_FONTDB);
 
+        // ratio to resize the image if needed
+        let resize_ratio = match resize {
+            None => 1.0,
+            Some(resize_struct) => calculate_resize(
+                resize_struct,
+                na::Vector2::new(svg_tree.size.width() as f64, svg_tree.size.height() as f64),
+                pos,
+            ),
+        };
```

Comment:
I think this calculation is better placed in the functions `generate_vector_image_from_bytes()`. Otherwise the parameters get a bit unclear - there's size, but it won't be the actual size if `resize` is also passed in. And then we don't need the `Resize` struct, we can pass in the arguments the calculation depends on with descriptive names directly into `calculate_size()`

Reply:
I know it's a little weird.

I did it that way because I need to know the size of the svg and the image before calculating the ratio to make the image smaller. But that size only gets calculated in the `from_svg_str` and `from_image_bytes` respectively, after the string/bytes are read/parsed.

If I want to change this calculation to be done one step before in the `import.rs` file, I need to move some of the logic in these functions, so that I either read/parse the data from the bytes there, calculate the ratio, and pass on the calculated size whilst reverting the changes on the `from_.._bytes` function (but then we effectively parse the images twice, because there are other calls to the functions that will push `str` and `bytes` to them directly,  hence I can't modify then to send as an argument the `Image` or `Tree` types directly without adding other modifications elsewhere).

I agree that for now the `Resize` + `size` combo doesn't make sense. 
I wonder if I could remove the `resize` variable and change `size` for an enum of type

```rs
enum size_or_resize { //terrible name, do you have a better idea ?
     Fixed(na::Vector2<f64>) // (1)
     Resized(Resize)
}
```
as all the calls that are resized are calls where the `size` would be `None` and that is also true in the other direction. That way the `Resize` struct has a clear meaning and doesn't class with the `size` parameter in any way

(1) : I can remove here the Option so that we either Resize to fit of give a fixed size, or leave it to be able to call with `Fixed(None)` and have the image be pasted like it was before, with a 1-1 pixel mapping.

Reply:
The position dependence is mostly there for the fixed vertical layout 

Basically I want it to prevent this : 
<img width="2304" alt="image" src="https://github.com/flxzt/rnote/assets/115779707/56ca3768-50e2-4bcb-9d52-e0ea66265ff4">

Still technically in the viewport (for the size) but slightly annoying to resize without changing mode, resizing then going back to a fixed layout.

I might be wrong (going to double check if I'm mistaken about this happening without the additional `layout_fixed` thing

Reply:
For the other modes, position doesn't really matter if the size is roughly similar to the viewport's size

Reply:
I've fixed the main thing with the `enum` and that cleans up code in the other session by quite a bit.

For positions, do you want me to change 
```rs
            (resize.max_viewpoint.x - pos_left_top_canvas.index(i)) / (current_width)
```
to
```rs
            (resize.max_viewpoint.x) / (current_width)
```

I kinda see the `position` thing as a kinda neat positionning thing. For relatively large images, just zoom to the image size you want and there you go it's going to be the size you want.  On the flipside, it won't do anything about smaller images.

For the `nudging` part, I kinda don't want to do it in this PR, because the idea is just a little too rough.
- If the canvas is really zoomed out, there's no reason to take these format borders into consideration, it's going to be way too small so there's another thing to check
- the threshold thing might be annoying, better to have it respect borders completly or not
- To make it clear to the user you'd have to add either toast messages the first time explaining it, add some graphic help somewhere to explain, and it would probably need a dedicated setting not to annoy people.

Maybe that's better to reserve this to a contextual or settings option somwhere :  "paste whilst respecting borders".